subsys/net/ip/net_if.c

```diff
@@ -4295,6 +4295,11 @@ static void update_operational_state(struct net_if *iface)
 	enum net_if_oper_state prev_state = iface->if_dev->oper_state;
 	enum net_if_oper_state new_state = NET_IF_OPER_UNKNOWN;
 
+	if (!device_is_ready(net_if_get_device(iface))) {
+		new_state = NET_IF_OPER_LOWERLAYERDOWN;
+		goto exit;
+	}
+
```

Comment:
Wouldn't it make sense to move this check after `net_if_is_admin_up()`? I think that if the interface was not brought up (or failed trying to do so) it should remain in `NET_IF_OPER_DOWN` state, regardless of the underlying device status. Or maybe even this part is not needed after all, since currently the inteface won't be brought ADMIN UP in case the underlying device is not ready.

Reply:
Yes, I agree, I will move the check after admin up check.